src/libs/ajax.js

```diff
@@ -471,43 +480,34 @@ const Billing = signal => ({
     return res
   },
 
-  billingProject: async projectName => {
-    const route = `billing/v2/${projectName}`
-    const res = await fetchRawls(route, _.merge(authOpts(), { signal, method: 'GET' }))
+  listProjectUsers: async projectName => {
+    const res = await fetchRawls(`billing/v2/${projectName}/members`, _.merge(authOpts(), { signal }))
     return res.json()
   },
 
-  project: projectName => {
-    const root = `billing/v2/${projectName}/members`
-
-    const removeRole = (role, email) => {
-      return fetchRawls(`${root}/${role}/${encodeURIComponent(email)}`, _.merge(authOpts(), { signal, method: 'DELETE' }))
-    }
-
-    const addRole = (role, email) => {
-      return fetchRawls(`${root}/${role}/${encodeURIComponent(email)}`, _.merge(authOpts(), { signal, method: 'PUT' }))
-    }
+  addProjectUser: (projectName, roles, email) => {
+    const addRole = role => fetchRawls(
+      `billing/v2/${projectName}/members/${role}/${encodeURIComponent(email)}`,
+      _.merge(authOpts(), { signal, method: 'PUT' })
+    )
 
-    return {
-      listUsers: async () => {
-        const res = await fetchRawls(root, _.merge(authOpts(), { signal }))
-        return res.json()
-      },
+    return Promise.all(_.map(addRole, roles))
+  },
 
-      addUser: (roles, email) => {
-        return Promise.all(_.map(role => addRole(role, email), roles))
-      },
+  removeProjectUser: (projectName, roles, email) => {
+    const removeRole = role => fetchRawls(
+      `billing/v2/${projectName}/members/${role}/${encodeURIComponent(email)}`,
+      _.merge(authOpts(), { signal, method: 'DELETE' })
+    )
 
-      removeUser: (roles, email) => {
-        return Promise.all(_.map(role => removeRole(role, email), roles))
-      },
+    return Promise.all(_.map(removeRole, roles))
+  },
 
-      changeUserRoles: async (email, oldRoles, newRoles) => {
-        if (!_.isEqual(oldRoles, newRoles)) {
-          await Promise.all(_.map(role => addRole(role, email), _.difference(newRoles, oldRoles)))
-          return Promise.all(_.map(role => removeRole(role, email), _.difference(oldRoles, newRoles)))
-        }
-      }
+  changeUserRoles: async (projectName, email, oldRoles, newRoles) => {
+    const billing = Billing(signal)
```

Comment:
This is a potentially destructive operation, so it shouldn't take `signal`, it should complete even if a modal or something is closed.

Reply:
Can you explain what this means? Why is this destructive?